src/FixedSizeList.js

```diff
@@ -51,14 +52,35 @@ const FixedSizeList = createListComponent({
         return maxOffset;
       case 'end':
         return minOffset;
-      case 'center':
-        return Math.round(minOffset + (maxOffset - minOffset) / 2);
+      case 'center': {
+        // "Centered" offset is usually the average of the min and max
+        // offsets. But near the beginning or end of the list, this math
+        // doesn't produce the actual closest-to-center offset, so we
+        // override.
+        const centered = Math.round(minOffset + (maxOffset - minOffset) / 2);
+        if (centered < Math.ceil(size / 2)) {
+          return 0; // near the beginning
+        } else if (centered > lastViewportOffset + Math.floor(size / 2)) {
+          return lastViewportOffset; // near the end
+        } else {
+          return centered;
+        }
+      }
       case 'auto':
       default:
         if (scrollOffset >= minOffset && scrollOffset <= maxOffset) {
           return scrollOffset;
-        } else if (scrollOffset - minOffset < maxOffset - scrollOffset) {
-          return minOffset;
+        } else if (scrollOffset < minOffset) {
+          // For auto alignment, if size isn't an even multiple of itemSize,
+          // always put the partial item at the end of the viewport because it
+          // looks better than at the beginning. Exception: don't do this if the
```

Comment:
Not sure I understand the reasoning behind this added complexity. Why does it "look better"?

Reply:
I think I disagree with this change. Probably going to remove it. It breaks the specification of "auto" scroll behavior, by over-scrolling in the case when you're scrolling from a lower to higher index.

Reply:
Imagine a list with `itemSize=30` and `size=100`.  Current behavior when scrolling forward (lower to higher index) with `align=auto` will look like `|--|------|------|--xx--|`, where 'xx' is the desired item and each `-` represents 5px.

I'm suggesting that `|------|------|--xx--|--|` is a better way to handle the case when items don't fit exactly into the viewport, because people generally read top-to-bottom (or left-to-right for horizontal LTR, or right-to-left for horiz RTL).  Users are used to seeing partial items at the end of a list, but seeing them at the start of a list is a somewhat jarring user experience that I was hoping to avoid. 

It's something of a judgement call.  I don't feel very strongly about this one way or the other, esp. because my current use of react-window doesn't trigger this case.  If you take it out I won't be offended. ;-)

BTW, I've been using the relatively-new CSS setting `scroll-snap-type` and it really improves UX for react-window by enforcing whole-item scrolling, so you're not left with partial items visible.

Reply:
Yeah, I think I understood the suggestion just not the reasoning behind it 😄 thanks for elaborating~

> BTW, I've been using the relatively-new CSS setting `scroll-snap-type` and it really improves UX for react-window by enforcing whole-item scrolling, so you're not left with partial items visible.

Ah, that's interesting. I haven't played with the two yet. I can see how it might be a UX improvement!

Reply:
Yep `scroll-snap-type` and `scroll-snap-align` have been big (and very easy) UX wins. Another nice one has been  and it massively improves UX. 

I've also had good results with `scroll-behavior: smooth;` which makes `scrollToItem` look really slick as the item slides gracefully into position. The only caveat with this one is that if you have an initial scroll offset that's large, then it smooth-scrolls on mount which can take 500-1000 msecs of items speeding by before it gets to the right item. IMHO this looks bad (and delays initial user interaction) so I've had to add this style using a `useEffect` after mounting.

It's amazing how much better browsers in general (and CSS in particular) have gotten in the last 5 years.  The surface area is huge though-- hard to keep up with!